### DISCUSSION TITLE
Is possible to pass an build.arg to a Dockerfile in a pipeline?

### DISCUSSION THREAD

**Discussion by tyoc213** [aws/copilot-cli#5403] [Q&A] [answered]
Hi there, I have a `Dockerfile` whicn one of its `RUN` lines is a clone of a private gitrepo (not when running, but on build phase), so something like `RUN git clone -b mybranch --single-branch https://${GIT_TOKEN}@github.com/something/something.git` and I normally deploy it like `GIT_TOKEN=XXX copilot svc deploy -n something -e environment`

But then I made/deployed a pipeline and when the pipeline is ran it always fails on step 2 of 3 as the value for `GIT_TOKEN` is null as printed on the logs of the `Dockerfile` which alongoside the clone has a `RUN echo "git token is ${GIT_TOKEN}"` and it only prints the line `git toke is` so no env var present.

Any way to do this?

I tried putting as other secrets

```
secrets:
  GIT_TOKEN: /copilot/${COPILOT_APPLICATION_NAME}/${COPILOT_ENVIRONMENT_NAME}/secrets/GIT_TOKEN
```

But still empty. Also `RUN env` on the `Dockerfile` doesnt show the desired GIT_TOKEN.

**Comment by KollaAdithya** [comment]
Hello @tyoc213 !

Looks like a similar ask mentioned in this [comment](https://github.com/aws/copilot-cli/issues/2090#issuecomment-1132880784)
Can you try this out if you want to pass `GIT_TOKEN` as secret.

Instead If you to pass env variables in pipeline as buildarg in the Dockerfile.

1. In pipeline `buildspec.yml` 
```
env:
  variables:
    GIT_TOKEN: "xyz"
```
2. In service manifest you get buildargs
```
  build:
    dockerfile: ./web/Dockerfile
    args:
     GIT_TOKEN: ${GIT_TOKEN}
```

**Reply by tyoc213** [reply]
Will give a shoot and some testing. 

But the thing is that I don't want to store the secret on git ever :), I have tried and indeed I can deploy if I just put the token there instead of ${GIT_TOKEN} but that doesn't work for security reasons

**Reply by tyoc213** [reply, marked as answer]
Well, at the end we copied first the repo we wanted to clone with GIT_TOKEN, so the argument was deleted from the call as we didnt found a way to make it with secrets of the environment and not put it on a commit.
